builder/qemu/step_prepare_efivars.go

```diff
@@ -0,0 +1,71 @@
+package qemu
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+)
+
+// stepPrepareEfivars copies the EFIVars file to the output, so we can boot
+// and use it as a RW flash drive
+type stepPrepareEfivars struct {
+	EFIEnabled  bool
+	OutputDir   string
+	EFIVarsPath string
+}
+
+const efivarStateKey string = "EFI_VARS_FILE_PATH"
+
+func (prepEfi *stepPrepareEfivars) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	ui := state.Get("ui").(packersdk.Ui)
+
+	if !prepEfi.EFIEnabled {
+		return multistep.ActionContinue
+	}
+
+	efiVarsOutput := fmt.Sprintf("%s/efivars.fd", prepEfi.OutputDir)
```

Comment:
nit: efiVarsOutput feels a bit repetitive here, as well. What do you think about just calling it `dst`, `targetPath` or `dstPath`?
```suggestion
	efiVarsOutput := fmt.Sprintf("%s/efivars.fd", prepEfi.OutputDir)
```

Regarding the construction of the file path do we need to worry about path separators? If you may want to use `filepath.Join` to construct the destination path.

Reply:
Good call for the `filepath.Join`.